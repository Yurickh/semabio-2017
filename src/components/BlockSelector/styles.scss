@import 'ambience/color';
@import 'ambience/responsive';

.block-selector {
  display: flex;
  justify-content: center;

  @include small-only {
    justify-content: space-around;
  }

  & > .block {
    font-size: 20px;

    margin-right: 42px;

    cursor: pointer;
    letter-spacing: .71px;

    color: #000;

    @include small-only {
      margin-right: 0;
    }
  }

  & > .block.-selected {
    border-bottom: $color-forest 3px solid;
  }
}
