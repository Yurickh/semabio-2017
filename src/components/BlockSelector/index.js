import React, { Component } from 'react'
import PropTypes from 'prop-types'

import './styles.css'

class BlockSelector extends Component {
  static propTypes = {
    currentBlock: PropTypes.number,
    onChange: PropTypes.func,
  }

  isSelected = (index) => {
    return this.props.currentBlock === index ? '-selected' : ''
  }

  selectBlock = (index) => {
    this.props.onChange(index)
  }

  render() {
    return (
      <div className="block-selector">
        {
          ['Bloco A', 'Bloco B', 'Bloco C', 'Bloco D', 'Bloco E'].map((block, index) => {
            return <div
              className={`block ${this.isSelected(index)}`}
              onClick={() => this.selectBlock(index)}
              key={block}
            >
              {block}
            </div>
          })
        }
      </div>
    )
  }
}

export default BlockSelector
