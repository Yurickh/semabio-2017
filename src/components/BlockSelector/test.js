import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import BlockSelector from '../BlockSelector'

describe('BlockSelector', () => {
  it('should render correctly', () => {
    const tree = shallow(<BlockSelector />)
    expect(toJSON(tree)).toMatchSnapshot()
  })
})
