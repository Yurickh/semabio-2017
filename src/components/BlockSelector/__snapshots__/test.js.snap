// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockSelector should render correctly 1`] = `
<div
  className="block-selector"
>
  <div
    className="block "
    onClick={[Function]}
  >
    Bloco A
  </div>
  <div
    className="block "
    onClick={[Function]}
  >
    Bloco B
  </div>
  <div
    className="block "
    onClick={[Function]}
  >
    Bloco C
  </div>
  <div
    className="block "
    onClick={[Function]}
  >
    Bloco D
  </div>
  <div
    className="block "
    onClick={[Function]}
  >
    Bloco E
  </div>
</div>
`;
